scripts/export.py

```diff
@@ -114,25 +113,29 @@
     for field in not_null_fields:
         assert flag_df[flag_df[field].isnull()].empty, f"{field} contains nulls"
 
-    assert flag_df[
-        ~flag_df[OUTLIER_TYPE_FIELD].isin(OUTLIER_TYPE_CODES.values())
-    ].empty, f"{OUTLIER_TYPE_FIELD} contains invalid codes"
-
-    assert flag_df[
-        (flag_df[IS_OUTLIER_FIELD] == "Y")
-        & (flag_df[OUTLIER_TYPE_FIELD] == OUTLIER_TYPE_CODES["Not outlier"])
-    ].empty, (
-        f"{OUTLIER_TYPE_FIELD} cannot be {OUTLIER_TYPE_CODES['Not outlier']} "
-        f"when {IS_OUTLIER_FIELD} is Y"
-    )
-
-    assert flag_df[
-        (flag_df[IS_OUTLIER_FIELD] == "N")
-        & (flag_df[OUTLIER_TYPE_FIELD] != OUTLIER_TYPE_CODES["Not outlier"])
-    ].empty, (
-        f"{OUTLIER_TYPE_FIELD} must be {OUTLIER_TYPE_CODES['Not outlier']} "
-        f"when {IS_OUTLIER_FIELD} is N"
-    )
+    for field in [OUTLIER_REASON1_FIELD, OUTLIER_REASON2_FIELD, OUTLIER_REASON3_FIELD]:
+        invalid_values = flag_df[~flag_df[field].isin(OUTLIER_TYPE_CODES.values())]
+        assert invalid_values.empty, f"{field} contains invalid codes"
+
+    # Tests confirming that a price outlier reason is needed
+    # for a sale to be counted as an outlier, and vice-versa
+    values_to_check = list(OUTLIER_TYPE_CODES.values())[1:5]
```

Comment:
**[Suggestion, non-blocking]** I think we'll thank ourselves later if we build this check slightly more defensively -- if we decide to change the order of the codes in the `OUTLIER_TYPE_CODES` dict, but we don't remember to update this line, we could end up changing the conditions of this check by accident. Slightly better might be to define a constant like so that could tie this check to the outlier codes themselves:

```python
PRICE_CODES = ["Low price", "High price", "Low price per square foot", "High price per square foot"]

...

mask_all_not = ~flag_df[columns_to_check].isin(PRICE_CODES).any(axis=1)
mask_any = flag_df[columns_to_check].isin(PRICE_CODES).any(axis=1)
```

It's possible that we'll change the outlier codes in the future, but I expect that if/when we do so we'll be feeling more sensitive to the possible side effects of the change than if we reordered the `OUTLIER_TYPE_CODES` dict. We could also add a quick check to alert us of this case:

```python
for price_code in PRICE_CODES:
    assert price_code in OUTLIER_TYPE_CODES, f"{price_code} is in PRICE_CODES but missing from OUTLIER_TYPE_CODES"
```